### PR TITLE
[minor] use dist.group.WORLD for default process group

### PR DESCRIPTION
**Pull request by min-xu-ai**
- this is slightly more efficient than the previous commit
  for get_process_group_cached.

## What does this PR do?
Fixes # (issue).

## Before submitting

- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [ ] Did you make sure to update the docs?
  - [x] N/A
- [ ] Did you write any new necessary tests?
  - [x] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [x] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: fairscale/utils/parallel.py [hunk 1]
```diff
@@ -87,15 +91,21 @@ def get_process_group_cached(ranks: Optional[List[int]] = None) -> ProcessGroup:
...
-    ranks_set = frozenset(ranks)  # take care of ordering and duplicates in the ranks list.
-    if ranks_set not in cache:
-        cache[ranks_set] = dist.new_group(list(ranks_set))
+    # Lookup and fill the cache if needed.
+    cache = get_process_group_cached._global_group_cache  # type: ignore
+    if ranks is not None:
+        # take care of ordering and duplicates in the ranks list.
+        ranks = sorted(list(set(ranks)))
+    if ranks not in cache:
+        cache[ranks] = dist.new_group(ranks)
```
Comment (anj-s) [inline fairscale/utils/parallel.py:109]: Given lists are unhashable how can we use it as a key here? I think `sorted` returns a list right?

Comment (min-xu-ai) [reply fairscale/utils/parallel.py]: great catch. The test also caught it. just fixed it.

Comment (min-xu-ai) [reply fairscale/utils/parallel.py]: push a better fix.

**Comment by min-xu-ai** [comment]
All tests are passing now.

File: fairscale/utils/parallel.py [hunk 2]
```diff
@@ -58,10 +58,14 @@ def enable_pytorch_sync_bn(module: torch.nn.Module) -> None:
             layer._specify_ddp_gpu_num(1)  # type: ignore
 
 
-def get_process_group_cached(ranks: Optional[List[int]] = None) -> ProcessGroup:
+def get_process_group_cached(ranks: Optional[Sequence[int]] = None) -> ProcessGroup:
     """
     Singleton PyTorch distributed group cache. Inspired by the code from fairseq.
 
+    Just like torch.distributed.new_group, this needs to be called on all ranks
+    at the same time when a new group is created, no matter the rank is a member of
```
Comment (anj-s) [inline fairscale/utils/parallel.py:66]: s/",no matter the rank..."/".This is true for all ranks irrespective of their group membership status." or some rephrasing of the last sentence?

**Review by anj-s — APPROVED** [review]

File: fairscale/utils/parallel.py (hunk 2 above)
Comment (min-xu-ai) [reply fairscale/utils/parallel.py:66]: Thank you! Good point. I will address and merge.

File: fairscale/utils/parallel.py [hunk 3]
```diff
@@ -58,10 +58,14 @@ def enable_pytorch_sync_bn(module: torch.nn.Module) -> None:
...
+    that group of not.
```
Comment (min-xu-ai) [inline fairscale/utils/parallel.py:67]:
```suggestion
    Just like torch.distributed.new_group, this method needs to be called on all ranks
    at the same time when a new group is created. This is true for all ranks irrespective
    of their group membership status.
```

**Pull request merged**